src/Exthost/Diagnostic.re

```diff
@@ -0,0 +1,52 @@
+[@deriving yojson({strict: false})]
+type json = {
+  startLineNumber: int,
+  endLineNumber: int,
+  startColumn: int,
+  endColumn: int,
+  message: string,
+  severity: int,
+  // TODO:
+  // source: string,
+  //code: string,
+  // relatedInformation: DiagnosticRelatedInformation.t,
+};
+
+type t = {
+  range: OneBasedRange.t,
+  message: string,
+  severity: int,
+  // TODO:
+  // source: string,
+  // code: string,
+  // relatedInformation: DiagnosticRelatedInformation.t,
+};
+
+let of_yojson = json => {
+  switch (json_of_yojson(json)) {
+  | Ok(ret) =>
+    let range =
+      OneBasedRange.create(
+        ~startLineNumber=ret.startLineNumber,
+        ~startColumn=ret.startColumn,
+        ~endLineNumber=ret.endLineNumber,
+        ~endColumn=ret.endColumn,
+        (),
+      );
+
+    Ok({
+      range,
+      message: ret.message,
+      severity: ret.severity,
+      // TODO:
+      //source: ret.source,
+      //code: ret.code,
+    });
```

Comment:
Why not use `ocaml-decoders` and avoid all this unnecessary conversion and duplicate types? IMO `deriving yojson` should be outright banned.

Reply:
Ah, I ported this over from existing code... switched to use the decoder strategy now, though.

Reply:
Ah, sorry, from the description I thought it was new. But good to get it converted anyway!